enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,425 @@
+- Migrating from CCM to ALBC would require either disruption to customer workloads (e.g., deleting and recreating the load balancer) or a significant engineering effort to re-architect the way router deployments, load balancers, and DNS are managed. This would involve enabling two ELBs in parallel for the same router deployment, gradually shifting traffic, and then deleting the old ELB.
+- Red Hat may need to continue supporting CCM indefinitely for customers unwilling to migrate, meaning this would not be a one-release migration. It would likely require supporting all configurations (CLB with CCM, NLB with CCM, NLB with ALBC) as well as the migration process in perpetuity.
+- The above points only scratch the surface of special cases, such as custom security groups, custom DNS configurations, or potential regressions when transitioning from CCM to ALBC.
+
+### **Day-2 operations to switch the default router to use ALBC/LBC**:
+
+This would require users to manually deploy and configure ALBC after cluster installation, which does not meet the requirement for an opt-in during initial cluster deployment.
+
+## Open Questions [optional]
+
+> WIP
+
+- Q: Is it required to create a KEP to the CCM changes?
+
+    A: [No][a1]. But we will need to document the feature in the CCM repo.
+
+- Q: Does CCM require to recreate the NLB when configuration is updated (e.g., `Unmanaged`)?
+
+- Q: Does CCM require to recreate the NLB when configuration is added?
+
+- Q: [Can we reduce the number of rules in the backend SG provided by installer/CAPA][q1]?
+
+
+
+[a1]: https://github.com/openshift/enhancements/pull/1802#discussion_r2101097973
+
+[q1]: https://github.com/openshift/enhancements/pull/1802/files#r2112104305
+
+## Test Plan
+
+> WIP/TBReviewed
+
+**cloud-provider-aws (CCM):**:
+
+- e2e service Load Balancer type NLB with Security Groups (SG) needs to be implemented in the CCM component (upstream)
+- Implement e2e tests for service Load Balancer type-LoadBalancer NLB updating/disabling cloud-config. These tests should verify:
+
+    - The creation of an NLB with a newly created and associated Security Group.
+    - The correct configuration of Ingress rules in the Security Group based on the service ports and loadBalancerSourceRanges.
+    - The configuration of Egress rules, ensuring they are appropriately restricted.
+    - The deletion of the Security Group when the service is deleted.
+
+## Graduation Criteria
+
+> TODO/TBD
+
+### Dev Preview -> Tech Preview
+
+N/A. This feature will be introduced as Tech Preview (TBReviewed).
+
+### Tech Preview -> GA
+
+The E2E tests should be consistently passing, and a PR will be created to enable the feature gate by default.
```

Comment:
Expand the FG added here https://github.com/openshift/api/pull/2354

Initially we've been asked to go directly to TP, but considering the impact of this change (default to SG) we are considering starting from DP. We are evaluating the velocity in upstream and how fast we can move it.